lib/elixir/lib/task.ex

```diff
@@ -341,8 +341,17 @@ defmodule Task do
   Each `enumerable` item is passed as argument to the given function `fun` and
   processed by its own task. The tasks will be linked to the current process,
   similarly to `async/1`.
-
-  See `async_stream/5` for discussion, options, and examples.
+  
+  ## Example
+  
+  Count the graphemes in each string asynchronously, then add the counts together using reduce.
+  
+      iex> strings = ["long string", "longer string", "there are many of these"]
+      iex> stream = Task.async_stream(strings, fn text -> text |> String.graphemes |> Enum.count end)
```

Comment:
Yeah, I was just trying to simulate a process that may be more complex. But if you want to avoid teaching _not best practices_ in the official docs, then we could do something more complex like this:

```elixir
def frequency(list_of_strings) do
  stream = Task.async_stream(list_of_strings, &parse/1) 

  Enum.reduce(stream,
              %{}, 
              fn {:ok, map}, acc -> 
                Map.merge(map, acc, fn _, v1, v2 -> v1 + v2 end) 
              end)
end

defp parse(text) do
  text
  |> String.replace(~r/\s|[,!?.]|\d/, "")
  |> String.downcase
  |> String.graphemes 
  |> Enum.group_by(fn x -> x end)
  |> Enum.map(fn {k,v} -> {k, Enum.count(v)} end)
  |> Enum.into(%{})
end
```